src/accessors/state.rs

```diff
@@ -225,14 +314,15 @@ pub mod code {
 pub mod history_index {
     use super::*;
     use crate::kv::{mdbx::MdbxTransaction, tables::BitmapKey};
+    use anyhow::format_err;
     use croaring::Treemap as RoaringTreemap;
 
     pub fn find_next_block<'db: 'tx, 'tx, K, TK, E, H>(
         tx: &'tx MdbxTransaction<'db, TK, E>,
         table: H,
         needle: K,
         block_number: BlockNumber,
-    ) -> anyhow::Result<Option<BlockNumber>>
+    ) -> anyhow::Result<(Option<BlockNumber>, bool)>
```

Comment:
This complexifies one of the primary accessors for a pretty niche case of state walking.

Please revert this optimization.

Reply:
Removed